src/classes/TrialDifficultyExtensionController.cls

```diff
@@ -0,0 +1,95 @@
+public with sharing class TrialDifficultyExtensionController {
+
+
+    private Trial__c currentTrial;
+    private Trial_Level__mdt[] trialLevel;
+
+    public TrialDifficultyExtensionController(ApexPages.StandardController controller) {
+        currentTrial = [
+                SELECT Id, Trial_Level__c
+                FROM Trial__c
+                WHERE Id = :ApexPages.currentPage().getParameters().get('id')
+        ];
+        trialLevel = new List<Trial_Level__mdt>();
+
+    }
+
+    /*
+        @Description: The method finds a metaData object by criteria that is set
+        from the current record page "Trial_Level__c" field;
+
+        @param: currentTrail - stores Id and Trial_Level__c of the current Trail__c's record;
+                trialLevel - stores all fields of the Trial_Level__mdt;
+
+        @Returns: method returns list of the searched fields of the metaData object;
+    */
+    public Trial_Level__mdt getTrailDetails() {
+
+        if (currentTrial.Trial_Level__c != null) {
+            {
+                trialLevel = [
+                        SELECT Advanced_hiking_equipment_required__c, Advanced_hiking_footwear_required__c, Colour__c, Description__c,
+                                Dogs_allowed__c, Experienced_hikers_only__c, Difficulty_Level__c, For_Beginners__c,
+                                Kids_allowed__c, Water_required__c
+                        FROM Trial_Level__mdt
+                        WHERE Label = :this.currentTrial.Trial_Level__c
+                        LIMIT 1
+                ];
+                if (trialLevel.size() == 1) {
```

Comment:
List.isEmpty() looks better than size() > 0 or size() == x. In this case you will or will not have a element in a list (LIMIT 1).
No matter what from a logical point of view it's ok :)